src/gt4py/next/ffront/past_passes/type_deduction.py

```diff
@@ -217,12 +217,12 @@ def visit_Call(self, node: past.Call, **kwargs):
                         f"'{new_kwargs['out'].type}'."
                     )
             elif new_func.id in ["minimum", "maximum"]:
-                if new_args[0].type != new_args[1].type:
+                if arg_types[0] != arg_types[1]:
```

Comment:
Unrelated fix, so please don't forget to mention it in the PR/commit description.